cabot/metricsapp/forms/elastic.py

```diff
@@ -22,3 +24,44 @@ def clean_urls(self):
             return input_urls
         except ConnectionError:
             raise ValidationError('Invalid Elasticsearch host url(s).')
+
+
+class ElasticsearchStatusCheckForm(StatusCheckForm):
+    class Meta:
+        model = ElasticsearchStatusCheck
+        fields = (
+            'name',
+            'source',
+            'queries',
+            'check_type',
+            'warning_value',
+            'high_alert_importance',
+            'high_alert_value',
+            'time_range',
+            'frequency',
+            'active',
+            'retries',
+        )
+
+        def __init__(self, *args, **kwargs):
+            ret = super(ElasticsearchStatusCheckForm, self).__init__(*args, **kwargs)
+            self.fields['source'].queryset = ElasticsearchSource.objects.all()
+            return ret
+
+    def clean_queries(self):
+        """
+        Make sure input queries are formatted correctly.
+        """
+        queries = self.cleaned_data['queries']
+        try:
+            query_list = json.loads(queries)
+        except ValueError:
+            raise ValidationError('Queries must be json-parsable')
+
+        for query in query_list:
+            try:
+                validate_query(query)
```

Comment:
This should ideally move to the model.save() - In the future, we may support cabot APIs. The validation functionality can be at the core.